packages/devtools_app/lib/src/debugger/evaluate.dart

```diff
@@ -172,8 +172,7 @@ class _ExpressionEvalFieldState extends State<ExpressionEvalField> {
   }
 
   Future<String> _getBestStringValue(InstanceRef response) async {
-    if (response.valueAsString != null &&
-        response.valueAsStringIsTruncated == false) {
+    if (response.valueAsString != null && !response.valueAsStringIsTruncated) {
```

Comment:
I think we might want to switch the logic to `response.valueAsStringIsTruncated != true` (not null and not true). Given that that's slightly convoluted to read in an expression, we might want an extension method to make this more readable (`InstanceRef.valueAsStringCheckTruncated`?). Not sure of a good name there, or if we should instead just clarify things in a comment above the expression.

Reply:
>  I assume we should only assume the value is truncated if response.valueAsStringIsTruncated == true (i.e., not null, and equal to true)? I'll verify the logic in the PR.

It seems to be the case with my experiments.

> I think we might want to switch the logic to response.valueAsStringIsTruncated != true (not null and not true). 

I would like to say "null or false" instead, so I think the following would be the simplest  and clear - is this ok with you?

` response.valueAsStringIsTruncated == null  || response.valueAsStringIsTruncated == false `

> in terms of the crash, the client probably shouldn't call toString() if it has a non-truncated string value, however, it should still be able to call it (the backend should support the toString() call for all object types)

I agree, the crash looks like a bug. It happened because the VM proxy (dwds) couldn't find the primitive object by its reference. I am not sure what the reason is exactly, but I suspect that the object was simulated by the debugger and did not actually exist in chrome. We could fix this by cashing fake objects, but it does not look needed for values that have a string value, and the fix is good anyway to eliminate an unnecessary call. We can solve the problem of missing references separately, I will file a bug for webdev if I can recreate it for another case (like calling another functions on that object via invoke or evaluate API.

Reply:
I changed my mind -  `added response.valueAsStringIsTruncated != true` as you suggested  - as it does mean what I wanted to say, "null or false", and added a clarifying comment.